sqlite3/lib/src/wasm/function_store.dart

```diff
@@ -151,6 +151,8 @@ class FunctionStore {
   void _contextSetResult(Pointer context, Object? result) {
     if (result == null) {
       _bindings.sqlite3_result_null(context);
+    } else if (result is int) {
+      _bindings.sqlite3_result_int64(context, BigInt.from(result));
```

Comment:
I think this might not work? `int` and `double` are the same type on the web. So if I want to insert `3.5`, `result is int` would be true and we'd bind the wrong type.

Reply:
Under the hood Dart simply check it is fraction or not. So this type check order is good. But vice versa, if you check int type variable with `is double` it will always be true. 

```JavaScript
core.int = class int extends core.num {
    static is(o) {
      return typeof o == "number" && Math.floor(o) == o;
    }
}
core.double = class double extends core.num {
    static is(o) {
      return typeof o == "number";
    }
}
```

Reply:
Oh nice I didn't know about that, I thought the types were identical. To ensure this is working, can we add a test to ensure that `SELECT typeof(?)` executed with an integer as a parameter evaluates to `integer` on all platforms? Could be in `test/common/database.dart`.

Reply:
I can't wait for for dart2wasm 😄 https://github.com/dart-lang/sdk/commit/6faa5f3bd00ad8cbc640b3fc80cf7466c002a7df